automated install/basic-install.sh

```diff
@@ -301,8 +301,13 @@ package_manager_detect() {
         local phpVer="php"
         if is_command php ; then
             printf "  %b Existing PHP installation detected : PHP version %s\\n" "${INFO}" "$(php <<< "<?php echo PHP_VERSION ?>")"
-            printf -v phpInsMajor "%d" "$(php <<< "<?php echo PHP_MAJOR_VERSION ?>")"
-            printf -v phpInsMinor "%d" "$(php <<< "<?php echo PHP_MINOR_VERSION ?>")"
+            printf -v phpInsMajor "%s" "$(php <<< "<?php echo PHP_MAJOR_VERSION ?>")"
+            printf -v phpInsMinor "%s" "$(php <<< "<?php echo PHP_MINOR_VERSION ?>")"
+            if [[ "$phpInsMajor" =~ [^[:digit:]] || "$phpInsMinor" =~ [^[:digit:]]  ]]; then
+                 printf "  %b No valid PHP version detected\\n" "${CROSS}"
+                # so exit the installer
+                exit
+            fi
```

Comment:
`PHP_VERSION` versions should always be in the format of `1.1x.1x`. 

I think a good way to structure it is:

```pseudo
if is_command php then
  PHP_VERSION = $(php <<< "<?php echo PHP_VERSION ?>")"
  if first char of PHP_VERSION is not numeric then
   print an error message that PHP is not functioning correctly
   print what we see is the PHP_VERSION (most likely a null or empty string but might have the error message returned by `php`.
   print an explanation that this is a fatal error and we are going to quit.
   exit
 else:
  do all the good stuff(TM)
 endif
```

Reply:
> `if first char of PHP_VERSION is not numeric then`

I'll do it the other way round, because it's easier to define what's valid

Reply:
SGTM.

I don't know why I always test negations instead of the condition.